chain/chain/src/chain.rs

```diff
@@ -4765,6 +4785,27 @@ impl<'a> ChainUpdate<'a> {
             }
         }
 
+        // TODO: fill in the logic here for which shards we need to update flat storage for
```

Comment:
small request: can you replace all the TODOs that you're adding with TODO(github_issue_id) ? it will make it easier to find the once we miss in the future.

Reply:
Will do in a separate PR